### PR TITLE
Improve Changeset error message when it as only one association

### PR DESCRIPTION
**Pull request by eksperimental**
It is kind of confusing showing "Did you mean one of ....?"
when there is only one suggestion.

### CODE REVIEW

**Comment by josevalim** [comment]
Can you please revert the formatter changes to the file? We haven't run the formatter yet. Thank you!

**Comment by eksperimental** [comment]
> Can you please revert the formatter changes to the file? We haven't run the formatter yet. Thank you!

Sorry, that was done automatically by my text editor.

File: lib/ecto/changeset.ex [hunk 1]
```diff
@@ -2917,8 +2917,11 @@ defmodule Ecto.Changeset do
 
   defp raise_invalid_assoc(types, assoc) do
     associations = for {_key, {:assoc, %{field: field}}} <- types, do: field
-    raise ArgumentError, "cannot add constraint to changeset because association `#{assoc}` does not exist. " <>
-                         "Did you mean one of `#{Enum.join(associations, "`, `")}`?"
+    one_of = if Enum.count(associations) == 1, do: "", else: "one of "
```
Comment (josevalim) [inline lib/ecto/changeset.ex:2920]:
```suggestion
    one_of = if match?([_], associations), do: "", else: "one of "
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**